lib/widgets/inset_shadow_box.dart

```diff
@@ -0,0 +1,45 @@
+import 'package:flutter/material.dart';
+
+/// A widget that overlays inset shadows on a child.
```

Comment:
Could you expand on this a little bit, maybe mentioning the scroll-through-shadow use case?

There's a similar-sounding effect that this looks like it _might_ describe, but it turns out it doesn't; that's `box-shadow: inset`. It would be nice to have a full implementation of `box-shadow: inset`—we have a TODO in the reaction-chip UI:

```dart
      // TODO shadow effect, following web, which uses `box-shadow: inset`:
      //   https://developer.mozilla.org/en-US/docs/Web/CSS/box-shadow#inset
      //   Needs Flutter support for something like that:
      //     https://github.com/flutter/flutter/issues/18636
      //     https://github.com/flutter/flutter/issues/52999
      //   Until then use a solid color; a much-lightened version of the shadow color.
      //   Also adapt by making [borderUnselected] more transparent, so we'll
      //   want to check that against web when implementing the shadow.
      bgUnselected: const HSLColor.fromAHSL(0.08, 210, 0.50, 0.875).toColor(),
```

—but that's not what `InsetShadowBox` is. 🙂 (`InsetShadowBox` wouldn't handle the "stadium" shape of the reaction chips, for example.)

Reply:
Makes sense. I think this means that we should rename this widget to `RectangularInsetShadowBox`, to make it clearer.